src/Shared/Json/JsonSerializerExtensions.cs

```diff
@@ -18,6 +18,7 @@ public static bool ShouldUseWith(this JsonTypeInfo jsonTypeInfo, [NotNullWhen(fa
 
     public static JsonTypeInfo GetReadOnlyTypeInfo(this JsonSerializerOptions options, Type type)
     {
+        options.TypeInfoResolver ??= new EmptyJsonTypeInfoResolver();
```

Comment:
Looks good. You can avoid creating a custom resolver if you just use `JsonTypeInfoResolver.Combine()` with no arguments.